packages/web-sdk/src/transports/batch/transport.ts

```diff
@@ -0,0 +1,54 @@
+import { BaseTransport, TransportItem, VERSION } from '@grafana/faro-core';
+
+import type { BatchTransportOptions } from './types';
+
+const DEFAULT_BATCH_SEND_TIMEOUT_MS = 250;
+const DEFAULT_BATCH_SEND_COUNT = 50;
+
+export class BatchTransport extends BaseTransport {
+  readonly name = '@grafana/faro-web-sdk:transport-batch-wrapper';
+  readonly version = VERSION;
+
+  private readonly batchSendCount: number;
+  private readonly batchSendTimeout: number;
+
+  private signalCount = 0;
+  private timeoutId?: number = undefined;
+
+  private signalBuffer: TransportItem[] = [];
+
+  constructor(private transport: BaseTransport, options: BatchTransportOptions) {
+    super();
+
+    this.batchSendCount = options.batchSendCount ?? DEFAULT_BATCH_SEND_COUNT;
+    this.batchSendTimeout = options.batchSendTimeout ?? DEFAULT_BATCH_SEND_TIMEOUT_MS;
+
+    // Send batched/buffered data when user navigates to new page, switches or closes the tab, minimizes or closes the browser.
+    // If on mobile, it also sends data if user switches from the browser to a different app.
+    document.addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'hidden') {
+        this.flush();
+      }
+    });
+  }
+
+  send(item: TransportItem): void {
+    clearTimeout(this.timeoutId);
```

Comment:
so if items keep being added at regular interval < 250s, it might not flush until buffer is full. Maybe it should force flush after max X milliseconds, even if items are being added?

Reply:
+1. I am used to batching being time framed and not reset with each extra send request

Reply:
Very good catch!

I'll the same config as Otel does. `send_batch_max_size`

From teh docs:
>send_batch_max_size (default = 0): The upper limit of the batch size. 0 means no upper limit of the batch size. This property ensures that larger batches are split into smaller units. It must be greater than or equal to send_batch_size.

Reply:
Is this resolved? I see that the timeout is reset in each send request. Or maybe Github is not good with changes :(

Reply:
Oooh thanks Kostas. 

Yes it is resolved with this change:
```
if (this.batchSendTimeout > 0) {
      window.setTimeout(() => this.flush(), this.batchSendTimeout);
    }
```

But I forgot to remove the left over `clearTimeout` at the top of  `send()` 🙈 
Will clean this up